### PR TITLE
DynamoDB and DAX resources

### PR DESCRIPTION
**Pull request by disq**
- [x] Check if the fetched columns/data is enough/easy to build the policies.

### CODE REVIEW

File: resources/dax_clusters.go
Comment (roneli) [inline resources/dax_clusters.go:33]: missing description

File: resources/dax_clusters.go
Comment (roneli) [inline resources/dax_clusters.go:18]: missing PK

File: resources/dynamodb_tables.go
Comment (roneli) [inline resources/dynamodb_tables.go:18]: Missing PK, delete filter and multiplex

File: resources/dynamodb_tables.go
Comment (roneli) [inline resources/dynamodb_tables.go:17]: maybe gives this a better description, you can override with description attribute

**Review by roneli — CHANGES_REQUESTED** [review]
Some review comments.

File: resources/dax_clusters.go
Comment (disq) [inline resources/dax_clusters.go:280]: Question: Keep this or make it an array?

File: resources/dynamodb_tables.go
Comment (disq) [inline resources/dynamodb_tables.go:531]: Question: Keep this or make it a map of Name vs. Object(type,name)? Order shouldn't matter anyway.

File: resources/dynamodb_tables.go
Comment (disq) [inline resources/dynamodb_tables.go:612]: Question: Keep this or make it a map of Name vs. Object(name,prov_throughput_override)? Order shouldn't matter anyway.

File: resources/dynamodb_tables.go
Comment (disq) [inline resources/dynamodb_tables.go:652]: Question: Keep this or make it a map of Name vs. simplifiedGlobalSecondaryIndex? Order shouldn't matter anyway.

File: resources/dax_clusters.go
Comment (irmatov) [reply resources/dax_clusters.go:280]: I would keep it as JSON. btw, why not simply encode `SecurityGroups` as JSON?

File: resources/dynamodb_tables.go
Comment (irmatov) [reply resources/dynamodb_tables.go:567]: I would simply encode `AttributeDefinitions` as JSON. But it's fine as is too. Map would allow easier retrieval of type by name.

File: resources/dynamodb_tables.go
Comment (irmatov) [reply resources/dynamodb_tables.go:648]: map could be useful, but for me it's fine either way

File: resources/dynamodb_tables.go
Comment (irmatov) [reply resources/dynamodb_tables.go:688]: ditto. map could be more useful.

**Review by irmatov — APPROVED** [review]
LGTM

File: resources/dax_clusters.go
Comment (disq) [reply resources/dax_clusters.go:280]: Since the original types don't have json structs, the resulting keys would not be camel case and they might be very long. `{"SecurityGroupIdentifier": "theidentifier", ...}` etc. But this way we can just have `identifier` instead.

File: resources/dynamodb_tables.go
Comment (roneli) [inline resources/dynamodb_tables.go:195]: table_size_bytes -> size_bytes  table_status -> status

File: resources/dynamodb_tables.go
Comment (roneli) [inline resources/dynamodb_tables.go:174]: Renames: 

table_class_last_update 

table_class

**Review by roneli — APPROVED** [review]
LGTM, minor renames

**Comment by disq** [comment]
Do not merge - it doesn't seem to be able to read auto scaling settings if the table is not a "global table". Investigating

**Comment by disq** [comment]
Looks like new resource is needed: `github.com/aws/aws-sdk-go-v2/service/applicationautoscaling`. It also encapsulates other types of resources too so it's best to add it as a new resource and fetch all autoscaling things.

File: client/client.go
Comment (roneli) [inline client/client.go:222]: don't see a need for the namespaces string, you can hardcode them in the multiplexer, and just use the WithAccountRegionNamespaces -> we should break it to WithAccount().WithRegion().WithNamespace(), maybe in a different PR.

File: client/multiplexers.go
Comment (roneli) [inline client/multiplexers.go:32]: see my comments in client.go

File: resources/applicationautoscaling_policies.go
Comment (roneli) [inline resources/applicationautoscaling_policies.go:62]:
```suggestion
			{
				Name:        "type",
				Description: "The scaling policy type.  This member is required.",
				Type:        schema.TypeString,
				Resolver:    schema.PathResolver("PolicyType"),
			},
```

File: resources/applicationautoscaling_policies.go
Comment (roneli) [inline resources/applicationautoscaling_policies.go:21]: are we sure ARN is enough here? guessing same policy can be on multiple resources in different namespaces?

File: resources/applicationautoscaling_policies.go
Comment (roneli) [inline resources/applicationautoscaling_policies.go:120]: you can pass them all to the channel no need 1 by 1

File: resources/dax_clusters.go
Comment (roneli) [inline resources/dax_clusters.go:251]: add a TODO here referencing #263, this should be automatic for all services in the account/region multiplexer

File: resources/dax_clusters.go
Comment (roneli) [inline resources/dax_clusters.go:264]: you can do res <- output.Clusters

File: resources/provider_test.go
Comment (roneli) [inline resources/provider_test.go:32]: this won't be required after change in client.go

File: client/config.go
Comment (roneli) [inline client/config.go:11]: I don't think this should be configured by the user.

File: client/client.go
Comment (roneli) [inline client/client.go:99]: move this to the multiplexer.go file.

File: client/client.go
Comment (roneli) [inline client/client.go:193]: would perhaps calls this AutoscalingNamespace since its specific to that.

**Review by roneli — CHANGES_REQUESTED** [review]
Looks good, def not an easy resource this one, added review comments.

File: client/client.go
Comment (disq) [reply client/client.go:222]: hardcoding led to failing tests because the multiplexer was multiplexing too much, maybe I missed something

File: resources/applicationautoscaling_policies.go
Comment (disq) [reply resources/applicationautoscaling_policies.go:20]: ARN contains the policy name as well. Real data:

```
arn                                          | arn:aws:autoscaling:us-east-1:<ACCOUNT>:scalingPolicy:<UUID>:resource/dynamodb/table/<TABLE>:policyName/DynamoDBReadCapacityUtilization:table/<TABLE>
name                                         | DynamoDBReadCapacityUtilization:table/<TABLE>
type                                         | TargetTrackingScaling
resource_id                                  | table/<TABLE>
scalable_dimension                           | dynamodb:table:ReadCapacityUnits
```

Comment (roneli) [reply resources/applicationautoscaling_policies.go:21]: Never trust AWS that is my motto

**Review by roneli — APPROVED** [review]
LGTM if all comments are resolved we can merge.

**Pull request merged**
